### PR TITLE
Fix Consumer compatibility with laravel v9.42.0 (#499)

### LINKED ISSUE #499: PHP Fatal error:  Declaration of VladimirYuldashev\LaravelQueueRabbitMQ\Consumer::stop($status = 0): int must be compatible with Illuminate\Queue\Worker::stop($status = 0, $options = null)
I used command: 

composer update at laravel version 9.42

log:

`PHP Fatal error:  Declaration of VladimirYuldashev\LaravelQueueRabbitMQ\Consumer::stop($status = 0): int must be compatible with Illuminate\Queue\Worker::stop($status = 0, $options = null) in /home/gabriel/Projekty/subjects_and_grades/vendor/vladimir-yuldashev/laravel-queue-rabbitmq/src/Consumer.php on line 200

   Symfony\Component\ErrorHandler\Error\FatalError 

  Declaration of VladimirYuldashev\LaravelQueueRabbitMQ\Consumer::stop($status = 0): int must be compatible with Illuminate\Queue\Worker::stop($status = 0, $options = null)

  at vendor/vladimir-yuldashev/laravel-queue-rabbitmq/src/Consumer.php:200
    196▕      *
    197▕      * @param  int  $status
    198▕      * @return int
    199▕      */
  ➜ 200▕     public function stop($status = 0): int
    201▕     {
    202▕         // Tell the server you are going to stop consuming.
    203▕         // It will finish up the last message and not send you any more.
    204▕         $this->channel->basic_cancel($this->consumerTag, false, true);

   Whoops\Exception\ErrorException 

  Declaration of VladimirYuldashev\LaravelQueueRabbitMQ\Consumer::stop($status = 0): int must be compatible with Illuminate\Queue\Worker::stop($status = 0, $options = null)

  at vendor/vladimir-yuldashev/laravel-queue-rabbitmq/src/Consumer.php:200
    196▕      *
    197▕      * @param  int  $status
    198▕      * @return int
    199▕      */
  ➜ 200▕     public function stop($status = 0): int
    201▕     {
    202▕         // Tell the server you are going to stop consuming.
    203▕         // It will finish up the last message and not send you any more.
    204▕         $this->channel->basic_cancel($this->consumerTag, false, true);

      +1 vendor frames 
  2   [internal]:0
      Whoops\Run::handleShutdown()
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 255
`

### PR DESCRIPTION
**Pull request by Arthur-Sk**
Closes #499

### CODE REVIEW

**Comment by dammy001** [comment]
Hey, @vladkucherov could you please merge this ASAP?

**Comment by sturtond** [comment]
Broke for me too

**Review by sturtond — APPROVED** [review]
Tested on Laravel 9.42

**Review by dammy001 — APPROVED** [review]

**Review by joao-lemes — APPROVED** [review]

**Review by GarryLai — APPROVED** [review]
Working on 9.42

**Review by vlazaay — APPROVED** [review]

**Review by hAz5 — APPROVED** [review]
tested and works on v9.42

**Review by hAz5 — APPROVED** [review]
tested and works on v9.42

**Review by sharjeelz — COMMENTED** [review]
perfect

**Review by sharjeelz — APPROVED** [review]

File: src/Consumer.php [hunk 1]
```diff
@@ -195,14 +195,15 @@ protected function daemonShouldRun(WorkerOptions $options, $connectionName, $que
      * Stop listening and bail out of the script.
      *
      * @param  int  $status
+     * @param  WorkerOptions|null  $options
      * @return int
      */
-    public function stop($status = 0): int
+    public function stop($status = 0, $options = null)
```
Comment (sjaakkie) [inline src/Consumer.php:201]: Re-add return type :int

**Review by sjaakkie — APPROVED** [review]

**Review by medeirosinacio — APPROVED** [review]

**Review by FelipeeDev — APPROVED** [review]

**Comment by medeirosinacio** [comment]
Hey, @vladkucherov could you please merge this ASAP?

**Comment by Lendy10** [comment]
would you please merge this ASAP mr. @vladkucherov?

**Comment by medeirosinacio** [comment]
I think @vladkucherov is not developing anymore, his github and twitter seem to be abandoned. This repositories itself has several open pull requests without response.

This is the best native implementation package with Laravel.

Could it be that we shouldn't work on a fork of this project and have more collaborators to help maintain it?

**Comment by sturtond** [comment]
> I think @vladkucherov is not developing anymore, his github and twitter seem to be abandoned. This repositories itself has several open pull requests without response.
> 
> This is the best native implementation package with Laravel.
> 
> Could it be that we shouldn't work on a fork of this project and have more collaborators to help maintain it?

I see he has no activity since July, 

I hope this isnt the case

**Review by SublateSamuel — APPROVED** [review]

**Comment by sharjeelz** [comment]
when we will have this merge?

**Review by SofLesc — APPROVED** [review]

**Comment by iamfarhad** [comment]
you can use of my laravel rabbitmq package 

https://github.com/iamfarhad/LaravelRabbitMQ

**Comment by amiranbarigreenweb** [comment]
I need this too

**Review by iamfarhad — APPROVED** [review]

**Review by amiranbarigreenweb — APPROVED** [review]

**Review by amiranbari — APPROVED** [review]

**Review by AliBayat — APPROVED** [review]
it works fine on 9.42 and above (tested also on 9.45)

**Review by vladkucherov — APPROVED** [review]

**Comment by vladkucherov** [comment]
Hey guys, I'm not the author of this package, nor a maintainer who can merge this PR.
I suppose you should reach @vyuldashev

**Comment by koAndy** [comment]
@vyuldashev please merge this!

**Comment by iamfarhad** [comment]
> @vyuldashev please merge this!

you can use my laravel rabbitmq package

https://github.com/iamfarhad/LaravelRabbitMQ

**Comment by M-Porter** [comment]
Resolved by https://github.com/vyuldashev/laravel-queue-rabbitmq/pull/502

**Pull request closed without merging**